codegen/tests/example.rs

```diff
@@ -18,38 +18,46 @@ impl World for MyWorld {
     }
 }
 
+#[given("non-regex")]
+fn test_non_regex_sync(w: &mut MyWorld) {
+    w.foo += 1;
+}
+
+#[given("non-regex")]
+async fn test_non_regex_async(w: &mut MyWorld, #[given(step)] ctx: &Step) {
+    time::sleep(Duration::new(1, 0)).await;
+
+    assert_eq!(ctx.value, "non-regex");
+
+    w.foo += 1;
+}
+
 #[given(regex = r"(\S+) is (\d+)")]
 async fn test_regex_async(
     w: &mut MyWorld,
     step: String,
-    #[given(context)] ctx: &StepContext,
+    #[given(step)] ctx: &Step,
```

Comment:
This better to be just `#[step]`. We can do that. While expanding `given` on the function, check it arguments and remove `#[step]` attribute from it after that.

Additionally, you should check for other `given`/`then`/`when` attrs to be present here, because only the last expansion should remove `#[step]` attribute from the argument.